pkg/compactor/retention/expiration.go

```diff
@@ -132,6 +133,43 @@ func NewTenantsRetention(l Limits) *TenantsRetention {
 	}
 }
 
+func (tr *TenantsRetention) PolicyFor(userID string, lbs labels.Labels) string {
+	policyStreamMapping := tr.limits.PoliciesStreamMapping(userID)
+	var (
+		matchedRule       *validation.PriorityStream
+		matchedPolicyName string
+		found             bool
+	)
+Outer:
+	for policyName, policyStreams := range policyStreamMapping {
+		for _, policyStream := range policyStreams {
+			for _, m := range policyStream.Matchers {
+				if !m.Matches(lbs.Get(m.Name)) {
+					continue Outer
```

Comment:
I think this is not correct. IIUC here if the labels do not match the matcher for this policyStream, we continue to the next policy instead of to the next policyStream within this policy.